File: Examples/R/tests/testParser.R
test_that("Parser exports construct and load TWD_land", {
  parserClasses <- c(
    "FMTparser", "FMTareaparser", "FMTlandscapeparser", "FMTactionparser",
    "FMTtransitionparser", "FMTconstantparser", "FMTlifespanparser",
    "FMTyieldparser", "FMToutputparser", "FMToptimizationparser",
    "FMTmodelparser", "FMTscheduleparser"
  )
  for (className in parserClasses) {
    expect_true(exists(className, inherits = TRUE))
  }

  primary <- fmtModelPath()
  parser <- new(FMTmodelparser)
  parser$setquietlogger()
  models <- parser$readproject(primary, c("LP"), TRUE, TRUE, TRUE)
  expect_length(models, 1L)
  expect_true(nzchar(models[[1]]$getname()))
})

File: Examples/Python/tests/testParser.py
from FMT import Parser


def testParserClassesAreExposedAndConstructible():
    constructors = {
        "FMTparser": Parser.FMTparser,
        "FMTareaparser": Parser.FMTareaparser,
        "FMTlandscapeparser": Parser.FMTlandscapeparser,
        "FMTactionparser": Parser.FMTactionparser,
        "FMTtransitionparser": Parser.FMTtransitionparser,
        "FMTconstantparser": Parser.FMTconstantparser,
        "FMTlifespanparser": Parser.FMTlifespanparser,
        "FMTyieldparser": Parser.FMTyieldparser,
        "FMToutputparser": Parser.FMToutputparser,
        "FMToptimizationparser": Parser.FMToptimizationparser,
        "FMTmodelparser": Parser.FMTmodelparser,
        "FMTscheduleparser": Parser.FMTscheduleparser,
    }
    for name, constructor in constructors.items():
        assert constructor() is not None


def testTwdLandProjectLoads(modelPath):
    parser = Parser.FMTmodelparser()
    parser.setquietlogger()
    models = parser.readproject(str(modelPath), ["LP"])
    assert len(models) == 1
    assert models[0].getname() is not None
